horovod/common/gloo/gloo_context.cc

```diff
@@ -140,6 +154,55 @@ void GlooContext::Initialize(const std::string& gloo_iface) {
     LOG(DEBUG) << "no rendezvous server provided, assuming single process execution";
   }
 
+  bool elastic = GetBoolEnvOrDefault(HOROVOD_ELASTIC, false);
```

Comment:
True, but for non-elastic we still want to initialize via env vars, as the code paths for doing it dynamically are particular to the ElasticDriver.  We could refactor that, but it would be a big change, better left to an optional follow-up PR.

Reply:
https://github.com/horovod/horovod/issues/1875